src/Extractor/HttpClient.php

```diff
@@ -198,8 +198,8 @@ public function fetch(UriInterface $url, bool $skipTypeVerification = false, arr
         }
```

Comment:
The whole heuristics above is prety ugly. What if there is no `<head>` tag? What if the `<html>` tag contains an attribute that includes a character not listed above. I wonder if it were not cleaner to just remove all downlevel-hidden conditional comments including their contents, and then remove all downlevel-revealed comments. Or just remove all HTML comments completely.

Reply:
Removing all HTML comments would probably be easier

Reply:
Looking at the new regex, it should do the former, so the HTML stuff above should be redundant.

Reply:
done e13c8e952a06b2ea13db07fca2668b48c4afd95d

Reply:
Done e13c8e952a06b2ea13db07fca2668b48c4afd95d